### PR TITLE
Link cards serdes

### PR DESCRIPTION
**Pull request by qnkhuat**
Serdes for link cards -- the virtual dashcards that are used to link to entities from a dashboard.

A link card is defined inside `dashboard.visualization_settings` and have shape
```clojure
{:model "database" ;; could also be : table, card, model, dashboard, collection
 :id    1}
 ```

This PR makes sure we serdes the linked entities correctly.

Related https://github.com/metabase/metabase/issues/28246

- - -
This change is [Reviewable](https://reviewable.io/reviews/metabase/metabase/28321)

### CODE REVIEW

**Review by bshepherdson — APPROVED** [review]

File: src/metabase/models/serialization/util.clj [hunk 1]
```diff
@@ -345,6 +345,13 @@
 
 (declare ^:private mbql-deps-map)
 
+(def ^:private link-card-model->toucan-model-str
+  {"card"      "Card"
+   "dataset"   "Card"
+   "table"     "Table"
+   "database"  "Database"
+   "dashboard" "Dashboard"})
```
Comment (calherries) [inline src/metabase/models/serialization/util.clj:353]: missing collection again

Comment (calherries) [reply src/metabase/models/serialization/util.clj:353]: Should we reuse `link-card-model->toucan-model` and get the string from the model key? Then we wouldn't have to update two places to serialize another model

File: enterprise/backend/test/metabase_enterprise/serialization/v2/e2e/yaml_test.clj [hunk 2]
```diff
@@ -492,3 +498,108 @@
...
+             DashboardCard _                         {:dashboard_id           dashboard-id
+                                                      :visualization_settings (link-card-viz-setting "database" db-id)}
+             DashboardCard _                         {:dashboard_id           dashboard-id
+                                                      :visualization_settings (link-card-viz-setting "table" table-id)}
+             DashboardCard _                         {:dashboard_id           dashboard-id
+                                                      :visualization_settings (link-card-viz-setting "dashboard" dash-id)}
+             DashboardCard _                         {:dashboard_id           dashboard-id
+                                                      :visualization_settings (link-card-viz-setting "card" card-id)}
+             DashboardCard _                         {:dashboard_id           dashboard-id
+                                                      :visualization_settings (link-card-viz-setting "dataset" model-id)}]
```
[the remark this answers is not in the document]
Comment (qnkhuat) [reply enterprise/backend/test/metabase_enterprise/serialization/v2/e2e/yaml_test.clj:557]: added it!

**Review by calherries — APPROVED** [review]
Nice!

**Pull request merged**